kernel/src/log_segment.rs

```diff
@@ -104,13 +113,240 @@ impl LogSegment {
     }
 }
 
+pub(crate) struct LogSegmentBuilder<'a> {
+    fs_client: &'a dyn FileSystemClient,
+    log_root: &'a Url,
```

Comment:
maybe this should be `table_root` if we are appending `_delta_log` to it? I recall not liking how many of these paths were handled but that's another issue for me to dig up and we can fix later :)

Reply:
Good point. Changed it to be called `table_root`.